src/common/UserAvatarWithPresence.js

```diff
@@ -41,10 +41,9 @@ export default class UserAvatarWithPresence extends PureComponent<Props> {
 
   render() {
     const { avatarUrl, email, size, onPress, shape } = this.props;
-    const fullAvatarUrl = avatarUrl.get().toString();
 
     return (
-      <UserAvatar avatarUrl={fullAvatarUrl} size={size} onPress={onPress} shape={shape}>
+      <UserAvatar avatarUrl={avatarUrl} size={size} onPress={onPress} shape={shape}>
```

Comment:
OK, filed that as #4305 -- it provides a diagnosis for an existing bug that I've noticed the symptoms of for a while. :slightly_smiling_face: 

The easiest way to cleanly handle this might actually be to fix #4305 first, in the crude way where we just hardcode a scaling factor. Fixing that will mean getting explicit about what the "size" argument means, and systematic about what code is responsible for choosing it: so making the "stop double-processing" fix from this branch, and also probably making the size a required argument so it's always explicit at the call site. Then because that'll all be explicit, it should be easier to make the refactorings like this "UserAvatar: Take `AvatarURL` object instead of string" commit and have them clearly not change what sizes we use.

Reply:
Nice, it'll be good to fix that blurriness! I just posted a question about React Native's [`PixelRatio`](https://reactnative.dev/docs/0.62/pixelratio) API at https://github.com/zulip/zulip-mobile/issues/4305#issuecomment-731421333.